### PR TITLE
[Feature] Add pagination settings

### LINKED ISSUE #196: Add support to the pagination setting customization at the index level
⚠️ This issue is generated, which means the examples and the namings do not necessarily correspond to the language of this repository.
Also, if you are a maintainer, please add any clarification and instructions about this issue.

Related to:
- parent issue: https://github.com/meilisearch/integration-guides/issues/205
- engine issue: https://github.com/meilisearch/specifications/pull/157, https://github.com/meilisearch/meilisearch/issues/2495

----

Add a `pagination` index settings resource to manage customization of the maximum number of reachable documents during the search at the index level.

- Expose `pagination` object on `/indexes/:index_uid/settings` endpoints.
- Add `GET/PATCH/DELETE` - `/indexes/:index_uid/settings/pagination` endpoints.

The `pagination` object definition is made of the following properties:
- `maxTotalHits` type (`int`)

More information about what each field does could be [checked in the spec.](https://github.com/meilisearch/specifications/pull/117)

Check the other customization methods for reference like: `synonyms`, `searchableAttributes`, `filterableAttributes`.

TODO:
- [ ] Add the new methods to customize `pagination`
  - [ ] GET `/indexes/:index_uid/settings/pagination`
  - [ ] PATCH `/indexes/:index_uid/settings/pagination`
  - [ ] DELETE `/indexes/:index_uid/settings/pagination`
- [ ] Add tests

### PR DESCRIPTION
**Pull request by ahmednfwela**
Fixes #196
Depends on #263

- [x] Tests Added

### CODE REVIEW

**Comment by brunoocasali** [comment]
Can you fix the conflicts here?

**Comment by ahmednfwela** [comment]
Conflicts fixed and rebased on #263

File: lib/src/pagination_settings.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const _defaultMaxTotalHits = 1000;
+
+class PaginationSettings {
```
Comment (brunoocasali) [inline lib/src/pagination_settings.dart:3]:
```suggestion
class Pagination {

```

**Comment by ahmednfwela** [comment]
added tests, and rebased on #263

**Comment by brunoocasali** [comment]
Sorry @ahmednfwela, but can you rebase here? It is hard to make the review with multiple changes happening

**Review by brunoocasali — APPROVED** [review]
+1 @ahmednfwela thanks a lot! :D 

bors merge

**Pull request merged**